CHANGELOG.md

```diff
@@ -3,16 +3,86 @@ All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
+### Changed
+[#446](https://github.com/plotly/dash-table/pull/446)
+- Table API rework
+#### NEW
+    - `column.sort_as_null`: Allows sorting behavior customization.
+        Accepts an array of string, number or booleans.
+
+#### REMOVED
+    - `column.clearable`: Allows clearing the value of a dropdown cell.
+        Removed in favor of `dropdown_**` `clearable` nested property
+    - `column.options`
+        Removed. Redundant with `dropdown`.
+    - `pagination_settings`
+        Replaced by two props `page_current` and `page_size`.
+
+#### RENAMED
+    - `column_static_tooltip`
+        Renamed `tooltip`.
+    - `column_conditional_tooltips`
+        Renamed `tooltip_conditional`.
+    - `filter`
+        Renamed `filter_query`.
+    - `sort_type`
+        Renamed `sort_mode`.
+    - `derived_filter_structure`
+        Renamed to `derived_filter_query_structure`.
+
+#### MODIFIED
+    - `column.deletable`: Allows column deletion.
+        Now accepts a boolean or an array of booleans (for multi-line headers)
+    - `column.editable_name`: Allows column renaming.
+        Renamed to `column.renamable`
+        Now accepts a boolean or an array of booleans (for multi-line headers)
+    - `column.id`
+        Now accepts `string` only -- `number` column ids can be casted to string
+    - `n_fixed_columns`: Will fix columns to the left.
+        Renamed to `fixed_columns`
+        Now accepts an object { headers: boolean, data: number } instead of a number.
+        { headers: true } determines the number of columns to fix automatically.
+    - `n_fixed_rows`: Will fix rows to the top.
+        Renamed to `fixed_rows`
+        Now accepts an object { headers: boolean, data: number } instead of a number.
+        { headers: true } determines the number of rows to fix automatically.
+    -  `pagination_mode`
+        Renamed to `page_action`.
+    -  `column_static_dropdown`
+        Renamed `dropdown`.
+        Now an object with each entry refering to a Column ID. Each nested prop expects
+        `clearable` and `options`.
+    - `column_conditional_dropdowns`
+        Renamed to `dropdown_conditional`.
+        `condition` changed to the same `if` nested prop used by styles.
+        `dropdown` renamed to `options`.
+    - `dropdown_properties`
+        Renamed to `dropdown_data`.
+        Matches the `data` structure.
+    - `tooltips`
+        Renamed `tooltip_data`.
+        Matches the `data` structure.
+    - `filtering`
+        Renamed `filter_action`.
+    - `sorting`
+        Renamed `sort_action`.
+    - `sorting_treat_empty_string_as_none`
+        Renamed `sort_as_null`.
+        Now accepts an array of string, number or booleans.
+        Table-level prop for the `column.sort_as_null` column nested prop.
+    - `style_**_conditional`
```

Comment:
```suggestion
    - `style_data_conditional`
```

(It seems like that was the only one affected https://github.com/plotly/dash-table/pull/446/files#diff-6169e0f9bf18b8550e8a1a83425d56b8R966)